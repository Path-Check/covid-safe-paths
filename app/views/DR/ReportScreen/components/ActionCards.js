import React, { useContext } from 'react';
import { Image, View, Linking } from 'react-native';
import {
  Badge, Card, CardItem, Text, Left, Button, Right,
} from 'native-base';
import Icon from 'react-native-vector-icons/FontAwesome';
import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from 'react-native-responsive-screen';
import context from './reduces/context';
import styles from './styles';
import Colors from '../constants/Colors';

const { orange, green, mainBlue } = Colors;

export function Feels({ navigation }) {
  return (
    <View>
      <Card style={styles.bigCards}>
        <View
          style={{
            alignItems: 'center',
            height: wp('10%'),
            flexDirection: 'row',
          }}
        >
          <Icon name="heartbeat" color={orange} size={wp('7%')} />
          <Text style={[styles.textHeader, { marginLeft: 8 }]}>
            ¿Cómo te sientes?
          </Text>
        </View>
        <View style={styles.tester}>
          <Left>
            <Text style={styles.text}>
              Analizaremos tus síntomas para validar o descartar el COVID-19
            </Text>
          </Left>
          <Button
            onPress={() => navigation.navigate('Reportar')}
            style={[styles.buttons, { backgroundColor: green }]}
          >
            <Text style={styles.buttonText}>Evaluar</Text>
          </Button>
        </View>
      </Card>
    </View>
  );
}

export function LastBulletin() {
  const [globaState] = useContext(context);
  const { bulletins } = globaState;
  const lastBulletin = bulletins[0] || {};
  const { url } = lastBulletin;

  return (
    <View>
      <Card style={styles.bigCards}>
        <View
          style={{
            alignItems: 'center',
            height: wp('10%'),
            flexDirection: 'row',
          }}
        >
          <Icon name="clock-o" color={orange} size={wp('7%')} />
          <Text style={[styles.textHeader, { marginLeft: 8 }]}>
            ¿Quieres ver el último boletín?
          </Text>
        </View>
        <View style={styles.tester}>
          <Left>
            <Text style={styles.text}>
              Descarga el boletín oficial más reciente de COVID-19
            </Text>
          </Left>
          <Button
            onPress={() => (url ? Linking.openURL(url) : null)}
            style={[styles.buttons, { backgroundColor: green, marginLeft: 10 }]}
          >
            <Text style={styles.buttonText}>Ver</Text>
          </Button>
        </View>
      </Card>
    </View>
  );
}

export function Contact({ isProfile }) {
  const buttonStyle = isProfile
    ? {
      alignSelf: 'center',
      backgroundColor: '#0161f2',
      borderRadius: 25,
      justifyContent: 'center',
      height: hp('6%'),
      width: wp('27%'),
    }
    : styles.buttons;

  const buttonValueStyle = isProfile
    ? {
      color: 'white',
      fontSize: hp('2.1%'),
      fontWeight: '500',
      textAlign: 'center',
      letterSpacing: 0.9,
    }
    : styles.buttonText;

  return (
    <View>
      <Card
        style={[
          styles.bigCards,
          {
            backgroundColor: mainBlue,
            height: hp('15%'),
            justifyContent: 'center',
          },
        ]}
      >
        <View style={{ flexDirection: 'row' }}>
          <Left style={{ flexDirection: 'row' }}>
            <Image
              style={styles.image}
              // eslint-disable-next-line global-require
              source={require('../assets/images/logo_msp.png')}
            />
            <Left style={{ marginLeft: 15 }}>
              <Text style={[styles.text, { color: '#fff' }]}>
                Linea informativa
              </Text>
              <Text style={[styles.textHeader, { color: '#fff' }]}>*464</Text>
            </Left>
            <Button style={[buttonStyle, { backgroundColor: '#fff' }]}>
              <Text style={[buttonValueStyle, { color: '#0161F2' }]}>
                Chatear
              </Text>
            </Button>
          </Left>
        </View>
      </Card>
    </View>
  );
}

export function Alerts() {
  return (
    <View>
      <Card style={styles.bigCards}>
        <CardItem header>
          <Left>
            <Icon name="bell-o" color={mainBlue} size={30} />
            <Text style={styles.textHeader}>Alertas</Text>
          </Left>

          <Right style={{ backgroundColor: '#fff' }}>
            <Badge style={{ backgroundColor: mainBlue }}>
              <Text>9 +</Text>
            </Badge>
          </Right>
        </CardItem>
        <CardItem bordered>
          <Text>Sustituir con al data de la API</Text>
        </CardItem>
      </Card>
    </View>
  );
}
