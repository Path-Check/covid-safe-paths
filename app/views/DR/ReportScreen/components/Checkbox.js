import React, { useState, useEffect } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { CheckBox, Text } from 'native-base';
import Colors from '../constants/Colors';

export default function Checkbox({ id, text, setValue, initialCheck }) {
  const [checked, setChecked] = useState(initialCheck);

  useEffect(() => {
    setChecked(initialCheck);
  }, [initialCheck]);

  return (
    <View>
      <TouchableOpacity
        style={{ flexDirection: 'row', marginVertical: 8 }}
        onPress={() => {
          setChecked(!checked);
          setValue(id, !checked);
        }}>
        <CheckBox
          checked={checked}
          color={Colors.mainBlue}
          onPress={() => {
            setChecked(!checked);
            setValue(id, !checked);
          }}
        />
        <Text style={{ marginLeft: 15 }}>{text}</Text>
      </TouchableOpacity>
    </View>
  );
}
