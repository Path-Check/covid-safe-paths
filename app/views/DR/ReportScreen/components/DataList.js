import React from 'react';
import { widthPercentageToDP as wp } from 'react-native-responsive-screen';
import {
  View, TouchableOpacity, Text, StyleSheet,
  Image, Linking, ActivityIndicator,
} from 'react-native';

export default function DataList({
  data, styleTitle = {}, styleDescription = {}, styleDate = {},
}) {
  const isData = (data.length > 0);

  return isData ? (
    <>
      {data.map(({
        url = '#', image: { source }, title = '', date = '', content = '',
      }, index) => (
          <TouchableOpacity
            onPress={() => Linking.openURL(url)}
            key={String(index)}
            style={styles.itemContainer}
          >
            <Image style={styles.image} source={source} />
            <View style={styles.right}>
              <Text numberOfLines={1} style={[styles.title, styleTitle]}>{title}</Text>
              {content ? (
                <Text style={[styles.description, styleDescription]} numberOfLines={3}>
                  {content}
                </Text>
              ) : null}
              {date ? (
                <Text style={[styles.date, styleDate]}>
                  {date.toUpperCase()}
                </Text>
              ) : null}
            </View>
          </TouchableOpacity>
        ))}
    </>
  ) : (<ActivityIndicator size="large" />);
}

const styles = StyleSheet.create({
  itemContainer: {
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#fff',
    height: wp('20%'),
    borderRadius: 10,
    maxHeight: 90,
    marginVertical: 5,
    marginHorizontal: 10,
    padding: 5,
    elevation: 5,
    alignItems: 'center',
  },
  image: {
    height: wp('15%'),
    width: wp('15%'),
    borderRadius: 8,
  },
  description: {
    fontSize: 11,
    fontFamily: 'OpenSans-Regular',
  },
  title: {
    fontFamily: 'OpenSans-Bold',
    fontSize: 14,
  },
  date: {
    fontFamily: 'OpenSans-Regular',
    fontSize: 10,
    color: '#808080',
  },
  right: {
    width: wp('70%'),
    paddingHorizontal: 10,
  },
});
