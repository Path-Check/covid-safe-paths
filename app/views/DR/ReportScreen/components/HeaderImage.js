import { LinearGradient } from 'expo-linear-gradient';
import React from 'react';
import { ImageBackground, StyleSheet, Text } from 'react-native';
import { widthPercentageToDP as wp } from 'react-native-responsive-screen';

import placeholder from '../assets/images/placeholder.png';

export default function HeaderImage({ imgUrl, title }) {
  return (
    <ImageBackground
      loadingIndicatorSource={placeholder}
      resizeMode='cover'
      resizeMethod='scale'
      style={styles.container}
      source={imgUrl}>
      <LinearGradient
        colors={['rgba(0,0,0,0)', 'rgba(0,0,0,0.2)', 'rgba(0,0,0,0.6)']}
        style={[
          styles.container,
          styles.gradient,
          {
            flex: 1,
          },
        ]}>
        <Text style={styles.title}>{title}</Text>
      </LinearGradient>
    </ImageBackground>
  );
}

const styles = StyleSheet.create({
  container: { width: wp('100%'), height: wp('50%') },
  description: {
    fontSize: 11,
    color: '#fff',
    fontFamily: 'OpenSans-Regular',
  },
  title: {
    fontFamily: 'OpenSans-Bold',
    color: '#fff',
    fontSize: 17,
  },
  gradient: {
    justifyContent: 'flex-end',
    padding: 20,
  },
});
