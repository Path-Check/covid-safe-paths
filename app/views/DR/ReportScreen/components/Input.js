import React, { useState } from 'react';
import { TextInput, View } from 'react-native';
import styles from './styles';
import Colors from '../constants/Colors';

export default function Input({
  length,
  keyboard = 'default',
  onChange,
  multiLine,
  style,
  ...props
}) {
  const [focused, setfocused] = useState(false);
  const inputStyle = {
    borderColor: Colors.mainBlue,
    borderBottomWidth: 1
  };
  return (
    <View>
      <TextInput
        maxLength={length}
        keyboardType={keyboard}
        onFocus={() => setfocused(true)}
        onBlur={() => setfocused(false)}
        multiline={multiLine}
        style={[styles.inputs, focused && inputStyle, style]}
        onChangeText={text => onChange(text)}
        {...props}
      />
    </View>
  );
}
