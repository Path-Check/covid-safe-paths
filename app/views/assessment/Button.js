import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { Typography } from '../../components/Typography';

import { Colors } from '../../styles';

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   color?: string;
 *   onPress: () => void;
 *   title: string;
 *   disabled: boolean;
 * }>} */
export const Button = ({ color, onPress, title, disabled = false }) => {
  let backgroundColor = color
    ? color
    : disabled
    ? Colors.secondaryBackground
    : Colors.secondaryBlue;
  let textColor = color
    ? color
    : disabled
    ? Colors.disabledButtonText
    : Colors.white;
  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      testID='assessment-button'>
      <View
        style={[
          styles.cta,
          {
            backgroundColor: backgroundColor,
          },
        ]}>
        <Typography use='body1' style={[styles.ctaText, { color: textColor }]}>
          {title}
        </Typography>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  cta: {
    borderRadius: 10,
    paddingVertical: 22,
    paddingHorizontal: 14,
  },
  ctaText: {
    color: Colors.faintGray,
    textAlign: 'center',
  },
});
