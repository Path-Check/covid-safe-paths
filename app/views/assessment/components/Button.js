import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';

import { Typography } from '../../../components/Typography';

import { Colors } from '../../../styles';

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   textColor?: string;
 *   backgroundColor?: string;
 *   onPress: () => void;
 *   title: string;
 *   disabled: boolean;
 * }>} */
export const Button = ({
  textColor,
  backgroundColor,
  onPress,
  title,
  buttonStyle,
  textStyle,
  disabled = false,
}) => {
  const dynamicBackgroundColor = () => {
    if (backgroundColor) return backgroundColor;
    else if (disabled) return Colors.secondaryBackground;
    else return Colors.secondaryBlue;
  };

  const dynamicTextColor = () => {
    if (textColor) return textColor;
    else if (disabled) return Colors.disabledButtonText;
    else return Colors.white;
  };

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      testID='assessment-button'>
      <View
        style={[
          styles.cta,
          buttonStyle,
          { backgroundColor: dynamicBackgroundColor() },
        ]}>
        <Typography
          use='body1'
          style={[styles.ctaText, textStyle, { color: dynamicTextColor() }]}>
          {title}
        </Typography>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  cta: {
    borderRadius: 10,
    paddingVertical: 22,
    paddingHorizontal: 14,
  },
  ctaText: {
    color: Colors.faintGray,
    textAlign: 'center',
  },
});
