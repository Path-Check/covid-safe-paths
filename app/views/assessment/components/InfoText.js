import React from 'react';
import { StyleSheet } from 'react-native';
import { Typography } from '../../../components/Typography';

export const InfoText = ({
  titleStyle,
  descriptionStyle,
  useTitleStyle,
  useDescriptionStyle,
  title,
  description,
}) => {
  return (
    <>
      <Typography
        use={useTitleStyle}
        style={[styles.headingSpacing, titleStyle]}>
        {title}
      </Typography>
      {description && (
        <Typography
          use={useDescriptionStyle}
          style={[styles.description, descriptionStyle]}
          testID='description'>
          {description}
        </Typography>
      )}
    </>
  );
};

export const styles = StyleSheet.create({
  headingSpacing: {
    marginVertical: 30,
  },
  description: {
    marginBottom: 20,
  },
});
