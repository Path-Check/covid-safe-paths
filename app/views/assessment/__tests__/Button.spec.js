import { render } from '@testing-library/react-native';
import React from 'react';

import { Button } from '../components/Button';

test('base', () => {
  const { asJSON } = render(<Button title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});

test('disabled', () => {
  const { asJSON } = render(<Button disabled title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});

test('color', () => {
  const { asJSON } = render(<Button color='red' title='Next' />);
  expect(asJSON()).toMatchSnapshot();
});
