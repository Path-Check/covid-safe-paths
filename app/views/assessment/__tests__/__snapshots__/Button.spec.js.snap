// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 22,
          },
          undefined,
          Object {
            "backgroundColor": "#5061e6",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontFamily": "IBMPlexSans",
              "fontSize": 17,
              "lineHeight": 24,
            },
            Object {
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#f8f8f8",
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#ffffff",
              },
            ],
          ]
        }
      >
        Next
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`color 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={false}
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 22,
          },
          undefined,
          Object {
            "backgroundColor": "#5061e6",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontFamily": "IBMPlexSans",
              "fontSize": 17,
              "lineHeight": 24,
            },
            Object {
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#f8f8f8",
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#ffffff",
              },
            ],
          ]
        }
      >
        Next
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;

exports[`disabled 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    disabled={true}
    testID="assessment-button"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "paddingHorizontal": 14,
            "paddingVertical": 22,
          },
          undefined,
          Object {
            "backgroundColor": "#e5e7fa",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2e2e2e",
              "fontFamily": "IBMPlexSans",
              "fontSize": 17,
              "lineHeight": 24,
            },
            Object {
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "#f8f8f8",
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#a5affb",
              },
            ],
          ]
        }
      >
        Next
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
